client/src/app/services/auth.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { BehaviorSubject } from 'rxjs';
+import { AUTH_TOKEN, USER_ID } from './constants';
+
+
+@Injectable()
+export class AuthService {
+  private userId: string = null;
```

Comment:
you are assigning null to string. you can set this type to userId too or use empty string